src/test/java/com/amihaiemil/camel/ScalarTest.java

```diff
@@ -82,4 +84,25 @@ public void orphanForbidden() {
             new ArrayList<AbstractNode>(), "orphan"
         );
     }
+
+    /**
+     * Make sure that equals and hash code are reflexive,
+     * symmetric, and transitive.
+     */
+    @Test
+    public void testEqualsAndHashCode() {
```

Comment:
@salikjan try not to prefix or suffix test names with "test" - it's useless. Instead, Name tests after the 3rd person verbs. For instance this test I would name equalsAndHashCode().

See how I named the tests above: "hasNoChildren", "returns..." etc